pkg/controllers/workgenerator/controller.go

```diff
@@ -1015,14 +1154,23 @@ func (r *Reconciler) SetupWithManager(mgr controllerruntime.Manager) error {
 				newAvailableCondition := meta.FindStatusCondition(newWork.Status.Conditions, fleetv1beta1.WorkConditionTypeAvailable)
 
 				// we try to filter out events, we only need to handle the updated event if the applied or available condition flip between true and false
-				// or the failed placements are changed.
+				// or the failed/diffed/drifted placements are changed.
 				if condition.EqualCondition(oldAppliedCondition, newAppliedCondition) && condition.EqualCondition(oldAvailableCondition, newAvailableCondition) {
+					oldDriftedPlacements := extractDriftedResourcePlacementsFromWork(oldWork)
+					newDriftedPlacements := extractDriftedResourcePlacementsFromWork(newWork)
+					driftsEqual := utils.IsDriftedResourcePlacementsEqual(oldDriftedPlacements, newDriftedPlacements)
 					if condition.IsConditionStatusFalse(newAppliedCondition, newWork.Generation) || condition.IsConditionStatusFalse(newAvailableCondition, newWork.Generation) {
+						diffsEqual := true
+						if condition.IsConditionStatusFalse(newAppliedCondition, newWork.Generation) {
+							oldDiffedPlacements := extractDiffedResourcePlacementsFromWork(oldWork)
+							newDiffedPlacements := extractDiffedResourcePlacementsFromWork(newWork)
+							diffsEqual = utils.IsDiffedResourcePlacementsEqual(oldDiffedPlacements, newDiffedPlacements)
```

Comment:
why not compare the two works directly?

Reply:
I discussed with Chen and we had decided to keep the same process it already had. Do you think it would be better to do that instead?

Reply:
it is probably more efficient. I wonder why we wanted to keep it?

Reply:
For the sake of not changing the code too much. Will change to compare the two works directly.